### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by danielcardeenas**

### CODE REVIEW

File: src/controllers/browser.ts [hunk 1]
```diff
@@ -15,21 +15,27 @@ export async function initWhatsapp(session: string) {
...
-    path: require.resolve(path.join(__dirname, '../lib/middleware', 'middleware.js'))
+    path: require.resolve(
+      path.join(__dirname, '../lib/middleware', 'middleware.js')
+    ),
   });
 
   return page;
 }
 
 async function initBrowser(session: string) {
```
Comment (robertsLando) [inline src/controllers/browser.ts:29]: I suggest to add here the possibility to set the `headless` mode and the devTool `puppeteer` options, this is useful when someone wants to debug some problems (maybe some api change that throws an error on console) or simply when sulla is running in a background process and the user cannot see the QR. Based on this I also added an event to get the QR image data

File: src/lib/wapi/functions/send-message.js [hunk 2]
```diff
@@ -1,14 +1,12 @@
 export function sendMessage(id, message, done) {
-  var chat = WAPI.getChat(id);
-  if (chat) {
+  let chat = WAPI.getChat(id);
```
Comment (robertsLando) [inline src/lib/wapi/functions/send-message.js:2]: Is this actually working when the chat is not existing? I personally use WAPI `sendMessageToId` that fetches the contact from the contact list and sends a message to him

**Comment by robertsLando** [comment]
I even don't understand why you decided to make those changes on WAPI. I think this makes it harder to fix changes in case WAPI gets updated

File: src/controllers/browser.ts (hunk 1 above)
Comment (danielcardeenas) [reply src/controllers/browser.ts:39]: Adding

**Comment by danielcardeenas** [comment]
Having a file with more than 2000 lines is a really bad practice.
Keeping track of the original wapi.js is simple by just tracking the changes and git makes it easier.
Also, it is not my intention to depend fully on the original wapi.js we can debug it without a problem

File: src/lib/wapi/functions/send-message.js (hunk 2 above)
Comment (danielcardeenas) [reply src/lib/wapi/functions/send-message.js:2]: Exposed method to sulla now

**Pull request merged**